From ce4217d8ee9c28ba76ca34fae7a7c3ef1aceecb0 Mon Sep 17 00:00:00 2001
From: Guido Cella <guido@guidocella.xyz>
Date: Sat, 27 Jan 2024 22:43:29 +0100
Subject: [PATCH] cycle through tags

---
 shiftview.c  | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)
 create mode 100644 shiftview.c

diff --git a/shiftview.c b/shiftview.c
new file mode 100644
index 0000000..fa53db0
--- /dev/null
+++ b/shiftview.c
@@ -0,0 +1,19 @@
+// "arg->i" stores the number of tags to shift right (positive value)
+//          or left (negative value)
+void
+shiftview(const Arg *arg)
+{
+	Arg a;
+	int i = arg->i;
+	int nextseltags, curseltags = selmon->tagset[selmon->seltags];
+
+	if (i > 0) // left circular shift
+		nextseltags = (curseltags << i) | (curseltags >> (LENGTH(tags) - i));
+	else // right circular shift
+		nextseltags = curseltags >> (-i) | (curseltags << (LENGTH(tags) + i));
+
+	if (nextseltags & TAGMASK) {
+		a.i = nextseltags;
+		view(&a);
+	}
+}
-- 
2.52.0
