diff --git a/dwm.c b/dwm.c
index cee73b5..96e3c55 100644
--- a/dwm.c
+++ b/dwm.c
@@ -2390,6 +2393,26 @@ zoom(const Arg *arg) {
 	pop(c);
 }

+/** Function to shift the current view to the left/right
+ *
+ * @param: "arg->i" stores the number of tags to shift right (positive value)
+ *          or left (negative value)
+ */
+void
+shiftview(const Arg *arg) {
+	Arg shifted;
+
+	if(arg->i > 0) // left circular shift
+		shifted.ui = (selmon->tagset[selmon->seltags] << arg->i)
+		   | (selmon->tagset[selmon->seltags] >> (LENGTH(tags) - arg->i));
+
+	else // right circular shift
+		shifted.ui = selmon->tagset[selmon->seltags] >> (- arg->i)
+		   | selmon->tagset[selmon->seltags] << (LENGTH(tags) + arg->i);
+
+	view(&shifted);
+}
+
 int
 main(int argc, char *argv[]) {
 	if(argc == 2 && !strcmp("-v", argv[1]))


diff --git a/config.def.h b/config.def.h
index e47fff6..adb6a81 100644
--- a/config.def.h
+++ b/config.def.h
@@ -60,6 +60,8 @@ static const char *termcmd[]  = { "st", NULL };

 static Key keys[] = {
 	/* modifier                     key        function        argument */
+	{ MODKEY|ShiftMask,             XK_l,      shiftview,      { .i = +1 } },
+	{ MODKEY|ShiftMask,             XK_h,      shiftview,      { .i = -1 } },
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
